### DISCUSSION TITLE
PPM output support for RX

### DISCUSSION THREAD

**Discussion by FelixvdDonk** [ExpressLRS/ExpressLRS#1347] [Ideas]
The drone I'm targeting only supports PPM input. Is this an option that could be added or enabled in settings?

**Comment by Tygrys-1** [comment]
It is kinda not efficient. The PPM frame is more than 16ms long, so you will not benefit from ELRS rates of more than 50Hz.

**Comment by FelixvdDonk** [comment]
I understand, but I'm not looking for the benefits of high rates. I care about the long-range benefits. It's merely a problem of compatibility I'm trying to solve here

**Comment by DungAdumbs** [comment]
plenty of hardware sbus>ppm>pwm solutions out there.

**Comment by FelixvdDonk** [comment]
> 
> 
> plenty of hardware sbus>ppm>pwm solutions out there.

ExpressLRS doesnt output sbus. How is that a solution? @DungAdumbs

**Comment by DungAdumbs** [comment]
Very true, my oversight.  Have not yet seen a crsf hardware solution.

**Comment by microraptor** [comment]
Edit:
Sounds like SBUS support would be quite a lot of work and since it is a legacy protocol, there are no plans to implement it: #1182 

Old anwser:
SBUS is in development: https://github.com/ExpressLRS/ExpressLRS/issues/581#issuecomment-865076299
As mentioned there is plenty of hardware for SBUS > PWM and some can also output PPM.

**Comment by goatzillax** [comment]
Using our existing hardware without extra PPM hardware widgets would be really nice.

**Comment by jace26** [comment]
Same boat here. PPM out straight from the RX would be nice as my vehicle can only take in PPM and I don't want to mess with converters. Crossfire does PPM out and that's what we currently use, but would love to switch to ELRS. PPM is used as the primary input on some industry applications because its easier to handle without a processor for pass-through applications is my understanding.

**Comment by axljackass** [comment]
also same boat here, only possible solution i can see is PWM to PPM converter, 
planning to make a DIY board that will directly attach to the PWM outputs of RX in that way no cables are needed, only single cable for PPM, legacy flight controller boards still can be use neatly
found few working programs so it will not be issue anymore

**Comment by cheng3100** [comment]
Some rc car like mjx use one of its rx channel to output ppm to control the light to achieve left turn/right turn/forward/reverse lighting effects . The hardware of esp32 on the elrs rx should fully support one of its io to output ppm. So I think this should be a really usefully feature for elrs

**Comment by mha1** [comment]
It's the year 2026 and ELRS can output SBUS. Just use a SBUS to PPM converter.

**Reply by cheng3100** [reply]
Less is always meaningful. The PPM isn't about retro style; rather, it's a method that current RC car manufacturers are finding effective for controlling all the lights and linking them to motion. It only requires one I/O output, and because it doesn't handle motion control, the PPM frequency can be very low.

**Reply by goatzillax** [reply]
> It's the year 2026 and ELRS can output SBUS. Just use a SBUS to PPM converter.

Not sure if the phrasing is still valid for 2026, but this is what we used to call a hardware solution for a software problem.
